helix-term/src/commands.rs

```diff
@@ -1405,6 +1405,7 @@ fn _goto(
         let new_pos = lsp_pos_to_pos(doc.text(), definition_pos, offset_encoding);
         doc.set_selection(view.id, Selection::point(new_pos));
         align_view(doc, view, Align::Center);
+        push_jump(editor);
```

Comment:
This now pushes the jump twice, we push the previous position at the start of `_goto`

Reply:
If you're now expected to push a jump after jumping, then other places using `push_jump` need to be updated too.

Reply:
Duplicated jumps will not be recorded. The issue is current `gd` you can only `ctrl-o` and cannot go back to the original `gd` jump by `ctrl-i`. If `gd` already jump to the other place, I don't see why we shouldn't record it.